### PR TITLE
metabase-lib enhancements [WIP]

### PR DESCRIPTION
**Pull request by tlrobinson**
This wraps the MBQL clauses returned by StructuredQuery's `breakouts`, `aggregations`, and `filters` with objects that have additional useful methods.

```javascript
query.filters()[0].dimension().displayName()
query.filters()[0].remove().update();
```

Note that these are subclasses of Array, so they still work fine as raw MBQL:

```
JSON.stringify(query.filters()[0])
["=",["field-id",10],123]
```

### CODE REVIEW

File: frontend/test/metabase-lib/queries/StructuredQuery-filters.unit.spec.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  });
+  describe("isDimension", () => {
+    it("should return true for the same dimension", () => {
+      const f = q.filters()[0];
+      expect(f.isDimension(["field-id", ORDERS_TOTAL_FIELD_ID])).toBe(true);
+      expect(
+        f.isDimension(Dimension.parseMBQL(["field-id", ORDERS_TOTAL_FIELD_ID])),
+      ).toBe(true);
+    });
+    it("should return false for differe dimensions", () => {
```
Comment (kdoh) [inline frontend/test/metabase-lib/queries/StructuredQuery-filters.unit.spec.js]:
```suggestion
    it("should return false for different dimensions", () => {
```

File: frontend/test/metabase-lib/queries/StructuredQuery-filters.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      expect(f.operatorOptions()[0].name).toEqual("=");
+    });
+  });
+  describe("isOperator", () => {
+    it("should return true for the same operator", () => {
+      const f = q.filters()[0];
+      expect(f.isOperator("=")).toBe(true);
+      expect(f.isOperator(f.operatorOptions()[0])).toBe(true);
+    });
+    it("should return false for differe operators", () => {
```
Comment (kdoh) [inline frontend/test/metabase-lib/queries/StructuredQuery-filters.unit.spec.js]:
```suggestion
    it("should return false for different operators", () => {
```

File: frontend/src/metabase/visualizations/lib/utils.js [hunk 3]
```diff
@@ -121,13 +121,26 @@ export function computeSplit(extents, left = [], right = []) {
...
   count: t`Count`,
   sum: t`Sum`,
   distinct: t`Distinct`,
   stddev: t`Standard Deviation`,
 };
+const AGGREGATION_NAME_REGEX = new RegExp(
+  `^(${Object.keys(AGGREGATION_NAME_MAP).join("|")})(_\\d+)?$`,
+);
+
+export function getFriendlyName(column) {
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/lib/utils.js]: Are some of the changes from https://github.com/metabase/metabase/pull/9429/files#diff-24925d6209e59d8b6faeafcf4c035405R135 in here as well?

File: frontend/src/metabase/query_builder/containers/QueryBuilder.jsx [hunk 4]
```diff
@@ -195,6 +195,14 @@ export default class QueryBuilder extends Component {
       // "New Question" link was clicked
       this.props.initializeQB(nextProps.location, nextProps.params);
     }
+
+    // NOCOMMIT
+    window.question = nextProps.question;
```
Comment (kdoh) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.jsx]:
```suggestion

```

File: frontend/src/metabase/query_builder/containers/QueryBuilder.jsx [hunk 5]
```diff
@@ -195,6 +195,14 @@ export default class QueryBuilder extends Component {
...
+    window.query = nextProps.query;
```
Comment (kdoh) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.jsx]:
```suggestion
```

File: frontend/src/metabase/query_builder/containers/QueryBuilder.jsx [hunk 6]
```diff
@@ -195,6 +195,14 @@ export default class QueryBuilder extends Component {
       // "New Question" link was clicked
       this.props.initializeQB(nextProps.location, nextProps.params);
     }
+
+    // NOCOMMIT
```
Comment (kdoh) [inline frontend/src/metabase/query_builder/containers/QueryBuilder.jsx]:
```suggestion
```

**Review by kdoh — CHANGES_REQUESTED** [review]
Couple typos and some window references we should remove but once that's cleared up this looks good to me.

File: frontend/src/metabase/visualizations/lib/utils.js (hunk 3 above)
Comment (tlrobinson) [reply frontend/src/metabase/visualizations/lib/utils.js]: Ah, yeah I merged that branch in. I might revert that.

**Review by kdoh — APPROVED** [review]

**Comment by tlrobinson** [comment]
This was merged into `notebook-mode`

**Pull request closed without merging**